packages/location-field/src/index.tsx

```diff
@@ -46,8 +48,27 @@ function getOptionId(index: number): string {
   return `${optionIdPrefix}-${index}`;
 }
 
-// FIXME have a better key generator for options
-let optionKey = 0;
+function generateOptionId(feature, indexedOptionLookup) {
+  // Use a default id if the feature does not have an id.
+  let id = "000";
+
+  // User saved locations do not have an id, so use the label.
+  if (feature.displayName) {
+    id = feature.displayName.replace(/\s/g, "");
+  }
+
+  // Some geocoders will return an id. Use that if it exists.
+  if (feature.properties) {
+    id = feature.properties.id || feature.properties.label.replace(/\s/g, "");
+  }
```

Comment:
I am in favor of trying to look for a feature id first, then fallback on using the feature label. What do you think?

Reply:
I agree, isn't that what will happen here? Wont the `||` operator only come into play when there's no id in `properties`?
If you're talking about the user locations, I can add that check but in general I was not seeing any ids for those

Reply:
Sorry, I meant to fallback on `feature.displayName` if `feature.properties` is not provided!